daemon/daemon_test.go

```diff
@@ -200,6 +201,16 @@ func TestNotify(t *testing.T) {
 		t.Fatalf("expected non-expired sample to be passed to notifier")
 	}
 
+	// Test that all samples are pruned on non-recoverable error
+	metricsLog.WriteSampleNow(2)
+	metricsLog.WriteSampleNow(2)
+	notifier.ExpectError(notify.NonRecoverableError(fmt.Errorf("mocked")))
+	err = daemon.notify()
+	checkExpectedError(t, err, "non-recoverable notify error: mocked")
+	samples, _, _ = metricsLog.GetSamples()
+	if len(samples) != 0 {
+		t.Fatalf("expected all samples to be pruned")
+	}
```

Comment:
Would you like to add a test for `RecoverableError`?

Reply:
Test case added.